### PR TITLE
Improve the search algorithm

### LINKED ISSUE #8418: Content search improvements
### Description

The search results should better prioritize the results. When I type "columns" I'm not getting results related to the `columns` option which IMO is more desirable. In that case, the _ColumnSorting_ entries overshadow the result.

The same happens for `cell`, `data`, `filters` options, and probably more.

### PR DESCRIPTION
**Pull request by budnix**
### Context

The PR improves the search algorithm for Documentation by adding the possibility to prioritize the search results. Thanks to that the Handsontable options such as `columns`, `data`, `comments`, `cell`, `filter`, and `filters` are not dimmed by other, less relevant results.

_[skip changelog]_

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. fixes #8418
2.
3.

### Affected project(s):
- [ ] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: docs/.vuepress/config.js [hunk 1]
```diff
@@ -128,5 +128,10 @@ module.exports = {
     searchMaxGuidesSuggestions: 5,
     searchMaxAPISuggestions: 10,
     fuzzySearchDomains: ['Core', 'Hooks', 'Options'],
+    // The list modifies the search results position. When the search phrase matches the pages
+    // below, the search suggestions are placed before the rest results. The pages declared in
+    // the array at the beginning have the highest display priority.
+    APISearchDomainPriorityList: ['Options'],
+    guidesSearchDomainPriorityList: [],
   }
```
Comment (Siemienik) [inline docs/.vuepress/config.js:136]: what do you think about moving all search focused options to one object like:
```js
search:{
    placeholder: 'Search...',
    guidesMaxSuggestions: 5,
    apiMaxSuggestions: 10,
    fuzzySearchDomains: ['Core', 'Hooks', 'Options'],
    // The list modifies the search results position. When the search phrase matches the pages
    // below, the search suggestions are placed before the rest results. The pages declared in
    // the array at the beginning have the highest display priority.
    apiSearchDomainPriorityList: ['Options'],
    guidesSearchDomainPriorityList: [],
}
```

File: docs/.vuepress/theme/components/SearchBox.vue [hunk 2]
```diff
@@ -141,6 +152,12 @@ export default {
       const maxAPI = this.$site.themeConfig.searchMaxAPISuggestions || SEARCH_MAX_SUGGESTIONS;
       const maxGuides = this.$site.themeConfig.searchMaxGuidesSuggestions || SEARCH_MAX_SUGGESTIONS;
       const fuzzySearchDomains = this.$site.themeConfig.fuzzySearchDomains || [];
+      const APISearchDomainPriorityList = ([
```
Comment (Siemienik) [inline docs/.vuepress/theme/components/SearchBox.vue:155]: I would like to follow the rule that shortcuts with more than 2 letters are used same way as a word: `apiSearchDomainPriorityList`

File: docs/.vuepress/config.js (hunk 1 above)
Comment (budnix) [reply docs/.vuepress/config.js:136]: I'm ok with that 👍

Comment (budnix) [reply docs/.vuepress/config.js:136]: I'll prepare a fix

File: docs/.vuepress/theme/components/SearchBox.vue [hunk 3]
```diff
@@ -155,105 +172,82 @@ export default {
 
       // At first, search for the phrase in the API reference main categories
       for (let i = 0; i < pages.length; i++) {
-        if (resAPI.length >= maxAPI) {
```
Comment (Siemienik) [inline docs/.vuepress/theme/components/SearchBox.vue:158]: It has a negative performance impact. (I didn't test how large yet). If there aren't any objections, I would prefer to sort all pages through their priority during component or app initialization.  WDYT?

File: docs/.vuepress/config.js (hunk 1 above)
Comment (budnix) [reply docs/.vuepress/config.js:138]: Done 5a75498939a52c8681cd5c1eea52d8575e5080d6

File: docs/.vuepress/theme/components/SearchBox.vue (hunk 3 above)
Comment (budnix) [reply docs/.vuepress/theme/components/SearchBox.vue:158]: I checked that and there are ~500-700 items to search (after that two loops + inner loops)? IMO it's nothing for JS.

File: docs/.vuepress/config.js (hunk 1 above)
Comment (Siemienik) [reply docs/.vuepress/config.js:138]: Thank you

File: docs/.vuepress/theme/components/SearchBox.vue (hunk 3 above)
Comment (Siemienik) [reply docs/.vuepress/theme/components/SearchBox.vue:158]: Ok, let's keep it as is. 
BTW. If any performance issues happen further, we should remove two loops first.

**Review by Siemienik — APPROVED** [review]
I tested with several different configurations, All works perfect 💪

**Review by wszymanski — APPROVED** [review]
The code is lot more clear and it works good. Nice to see it 🎉

**Pull request merged**